src/mapgen/mapgen_fractal.cpp

```diff
@@ -153,21 +160,25 @@ void MapgenFractalParams::writeParams(Settings *settings) const
 
 int MapgenFractal::getSpawnLevelAtPoint(v2s16 p)
 {
-	bool solid_below = false;  // Dry solid node is present below to spawn on
-	u8 air_count = 0;  // Consecutive air nodes above the dry solid node
-	s16 seabed_level = NoisePerlin2D(&noise_seabed->np, p.X, p.Y, seed);
-	// Seabed can rise above water_level or might be raised to create dry land
-	s16 search_start = MYMAX(seabed_level, water_level + 1);
-	if (seabed_level > water_level)
-		solid_below = true;
-
-	for (s16 y = search_start; y <= search_start + 128; y++) {
-		if (getFractalAtPoint(p.X, y, p.Y)) {  // Fractal node
+	bool solid_below = false; // Fractal node is present below to spawn on
+	u8 air_count = 0; // Consecutive air nodes above a fractal node
+	s16 search_start = 0; // No terrain search start
+
+	// If terrain present, don't start search below terrain or water level
+	if (spflags & MGFRACTAL_TERRAIN) {
```

Comment:
This conditional branch depends on `noise_seabed`, so I'd rather check `if (seabed) {` to ensure it's REALLY only called if the pointer exists.

Reply:
Ok.

Reply:
Done.